msticpy/sectools/geoip.py

```diff
@@ -278,27 +293,139 @@ class GeoLiteLookup(GeoIpLookup):
+        force_update : bool, optional
+            Force update can be set to true or false. depending on it, new download request will be initiated.
+
+        Returns
+        -------
+        str
+            Returns the absoulte path of local maxmind geolite city database after control flow logic.
+
+        """
+        if db_folder is None:
+            db_folder = self._PKG_DIR
+        force_update = self._force_update
+        _list_of_db_paths = glob.glob(db_folder + "/*.mmdb")
+
+        if _list_of_db_paths:
+            _latest_db_path = max(_list_of_db_paths, key=os.path.getmtime)
+            print(f"Local Maxmind City Database already present in {db_folder}")
+            # Check for out of date DB file with latest available
+            reader = geoip2.database.Reader(_latest_db_path)
+            # Retrive date of maxind city database accessing build_epoch property of reader object.
+            _last_mod_time = datetime.utcfromtimestamp(reader.metadata().build_epoch)
+            _db_age = datetime.utcnow() - _last_mod_time
+            if _db_age > timedelta(30):
+                print(
+                    f"Latest local Maxmind City Database present is older than 30 days. Downloading new database to {db_folder}"
+                )
+                self.download_and_extract_gzip(self._MAXMIND_DOWNLOAD, db_folder)
+            else:
+                if force_update:
+                    print(
+                        f"Force update is set to True. Downloading new database to {db_folder}"
+                    )
+                    self.download_and_extract_gzip(self._MAXMIND_DOWNLOAD, db_folder)
+                else:
+                    print(
+                        "Existing local Maxmind City Database is downloaded in last 30 days so it will be used"
+                    )
+        else:
+            print(
+                f"No local Maxmind City Database found. Downloading new database to {db_folder}"
+            )
+            self.download_and_extract_gzip(self._MAXMIND_DOWNLOAD, db_folder)
+            _list_of_db_paths = glob.glob(db_folder + "/*.mmdb")
+            _latest_db_path = _list_of_db_paths[0]
+
+        return _latest_db_path
+
+    def __init__(self, db_folder: str = None, force_update: bool = False):
+        """Return new instance of GeoLiteLookup class.
+        
+        Parameters
+        ----------
+        db_folder: str, optional
+            Provide absolute path to the folder containing MMDB file (e.g. '/usr/home' or 'C:\maxmind'). 
+            If no path provided, it is set to download msticpy package installation directory.
+        force_update : bool, optional
+            Force update can be set to true or false. depending on it, new download request will be initiated.
+
+        """
+        if db_folder is None:
+            db_folder = self._PKG_DIR
+        self._force_update = force_update
+        self._dbpath = self.get_geolite_dbpath(db_folder, self._force_update)
```

Comment:
Make separate calls to get dbpath and check & update.
If there is no dbpath found or auto_update (True) you can go to check_and_update, which should return the latest_dbpath.
Surround call to check_and_update with try/except.
- if it threw an exception AND there was a previous dbpath - you can issue a warning (import warnings, warnings.warn(...) and just use the previous dbpath
- if an exception and no previous dbpath you should exit with an error - probably rethrow the exception you caught, since this will have the most relevant info about what went wrong. (just use "raise" with no args inside the except block - this will rethrow the same exception and keep the call stack intact.

Reply:
added option to toggle auto_update and separate call to check_and_update function and get_geoip_db_path to return latest_dbpath